packages/modules-sdk/src/loaders/module-provider-loader.ts

```diff
@@ -0,0 +1,86 @@
+import { MedusaContainer, ModuleProvider } from "@medusajs/types"
+import { isString, promiseAll } from "@medusajs/utils"
+import { Lifetime, asFunction } from "awilix"
+
+export async function moduleProviderLoader({
+  container,
+  providers,
+  registerServiceFn,
+}: {
+  container: MedusaContainer
+  providers: ModuleProvider[]
+  registerServiceFn?: (
+    klass,
+    container: MedusaContainer,
+    pluginDetails: any
+  ) => Promise<void>
+}) {
+  if (!providers?.length) {
+    return
+  }
+
+  await promiseAll(
+    providers.map(async (pluginDetails) => {
+      await loadModuleProvider(container, pluginDetails, registerServiceFn)
+    })
+  )
+}
+
+export async function loadModuleProvider(
+  container: MedusaContainer,
+  provider: ModuleProvider,
+  registerServiceFn?: (klass, container, pluginDetails) => Promise<void>
+) {
+  let loadedProvider: any
+
+  const pluginName = provider.resolve ?? provider.provider_name ?? ""
+
+  try {
+    if (isString(provider.resolve)) {
+      loadedProvider = await import(provider.resolve)
+    } else {
+      loadedProvider = provider.resolve
+    }
+  } catch (error) {
+    throw new Error(
+      `Unable to find plugin ${pluginName} -- perhaps you need to install its package?`
+    )
+  }
+
+  loadedProvider = (loadedProvider as any).default
```

Comment:
```suggestion
  loadedProvider = (loadedProvider as any).default ?? loadedProvider
```

in case the resolve is already the loaded plugin it might not have the default